File: php/examples/accounts/autofeedsettings/v1/GetAutofeedSettingsSample.php
<?php
/**
 * Copyright 2025 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
require_once __DIR__ . '/../../../../vendor/autoload.php';
require_once __DIR__ . '/../../../Authentication/Authentication.php';
require_once __DIR__ . '/../../../Authentication/Config.php';
// [START merchantapi_get_autofeed_settings]
use Google\ApiCore\ApiException;
use Google\Shopping\Merchant\Accounts\V1beta\Client\AutofeedSettingsServiceClient;
use Google\Shopping\Merchant\Accounts\V1beta\GetAutofeedSettingsRequest;

/**
 * This class demonstrates how to get the autofeed settings of a Merchant Center account.
 */
class GetAutofeedSettingsSample
{
    /**
     * Get the autofeed settings of a Merchant Center account.
     *
     * @param array $config The configuration data for authentication and account ID.
     * @return void
     */
    public static function getAutofeedSettingsSample(array $config): void
    {
        // Get OAuth credentials.
        $credentials = Authentication::useServiceAccountOrTokenFile();

        // Create options for the client.
        $options = ['credentials' => $credentials];

        // Create a client.
        $autofeedSettingsServiceClient = new AutofeedSettingsServiceClient($options);

        // Create the AutofeedSettings name.
        $name = "accounts/" . $config['accountId'] . "/autofeedSettings";

        // Call the API.
        try {
            // Prepare the request.
            $request = (new GetAutofeedSettingsRequest())
                ->setName($name);

            print "Sending get AutofeedSettings request:\n";
            $response = $autofeedSettingsServiceClient->getAutofeedSettings($request);

            print "Retrieved AutofeedSettings below\n";
            print_r($response);
        } catch (ApiException $e) {
            print $e->getMessage();
        }
    }

    /**
     * Helper to execute the sample.
     *
     * @return void
     */
    public function callSample(): void
    {
        $config = Config::generateConfig();
        self::getAutofeedSettingsSample($config);
    }
}


// Run the script
$sample = new GetAutofeedSettingsSample();
$sample->callSample();
// [END merchantapi_get_autofeed_settings]

File: python/examples/accounts/autofeedsettings/v1/get_autofeed_settings_sample.py
# -*- coding: utf-8 -*-
# Copyright 2025 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
"""A module to get AutofeedSettings."""

# [START merchantapi_get_autofeed_settings]
from examples.authentication import configuration
from examples.authentication import generate_user_credentials
from google.shopping.merchant_accounts_v1beta import AutofeedSettingsServiceClient
from google.shopping.merchant_accounts_v1beta import GetAutofeedSettingsRequest

_ACCOUNT = configuration.Configuration().read_merchant_info()


def get_autofeed_settings():
  """Gets the AutofeedSettings of a Merchant Center account."""

  # Gets OAuth Credentials.
  credentials = generate_user_credentials.main()

  # Creates a client.
  client = AutofeedSettingsServiceClient(credentials=credentials)

  # Creates name to identify the AutofeedSettings.
  name = "accounts/" + _ACCOUNT + "/autofeedSettings"

  # Creates the request.
  request = GetAutofeedSettingsRequest(name=name)

  # Makes the request and catches and prints any error messages.
  try:
    response = client.get_autofeed_settings(request=request)
    print("Retrieved AutofeedSettings below")
    print(response)
  except RuntimeError as e:
    print("Get AutofeedSettings request failed")
    print(e)


if __name__ == "__main__":
  get_autofeed_settings()
# [END merchantapi_get_autofeed_settings]
